# HG changeset patch
# User Rik <rik@octave.org>
# Date 1409370118 25200
#      Fri Aug 29 20:41:58 2014 -0700
# Node ID edf5d63c82e1fc837c2bad5d748c6be3a33beae8
# Parent  83b88e20e9c1da03512a5f44a0b3b8510167ba45
Deprecate strmatch function.

* scripts/deprecated/strmatch.m: Moved from scripts/strings.  Add
deprecation warning to code and to docstring.

* strings.txi: Remove from manual

* scripts/deprecated/module.mk: Add to build system in deprecated directory.

* scripts/strings/module.mk: Remove from build system in strings directory.

diff --git a/doc/interpreter/strings.txi b/doc/interpreter/strings.txi
--- a/doc/interpreter/strings.txi
+++ b/doc/interpreter/strings.txi
@@ -437,18 +437,16 @@ general regular expressions, the followi
 @DOCSTRING(index)
 
 @DOCSTRING(rindex)
 
 @DOCSTRING(strfind)
 
 @DOCSTRING(strjoin)
 
-@DOCSTRING(strmatch)
-
 @DOCSTRING(strtok)
 
 @DOCSTRING(strsplit)
 
 @DOCSTRING(ostrsplit)
 
 @DOCSTRING(strread)
 
diff --git a/scripts/deprecated/module.mk b/scripts/deprecated/module.mk
--- a/scripts/deprecated/module.mk
+++ b/scripts/deprecated/module.mk
@@ -1,13 +1,14 @@
 FCN_FILE_DIRS += deprecated
 
 deprecated_FCN_FILES = \
   deprecated/bicubic.m \
   deprecated/find_dir_in_path.m \
   deprecated/isstr.m \
-  deprecated/nfields.m
+  deprecated/nfields.m \
+  deprecated/strmatch.m
 
 FCN_FILES += $(deprecated_FCN_FILES)
 
 PKG_ADD_FILES += deprecated/PKG_ADD
 
 DIRSTAMP_FILES += deprecated/$(octave_dirstamp)
diff --git a/scripts/strings/strmatch.m b/scripts/deprecated/strmatch.m
rename from scripts/strings/strmatch.m
rename to scripts/deprecated/strmatch.m
--- a/scripts/strings/strmatch.m
+++ b/scripts/deprecated/strmatch.m
@@ -16,16 +16,21 @@
 ##
 ## You should have received a copy of the GNU General Public License
 ## along with Octave; see the file COPYING.  If not, see
 ## <http://www.gnu.org/licenses/>.
 
 ## -*- texinfo -*-
 ## @deftypefn  {Function File} {} strmatch (@var{s}, @var{A})
 ## @deftypefnx {Function File} {} strmatch (@var{s}, @var{A}, "exact")
+##
+## @code{strmatch} is deprecated and will be removed in Octave version 4.6.
+## Use @code{strncmp} (normal case), or @code{strcmp} (@qcode{"exact"} case),
+## or @code{regexp} in all new code.
+##
 ## Return indices of entries of @var{A} which begin with the string @var{s}.
 ## The second argument @var{A} must be a string, character matrix, or a cell
 ## array of strings.  If the third argument @qcode{"exact"} is not given, then
 ## @var{s} only needs to match @var{A} up to the length of @var{s}.
 ## Trailing spaces and nulls in @var{s} and @var{A} are ignored when matching.
 ##
 ## For example:
 ##
@@ -37,27 +42,33 @@
 ## strmatch ("apple", ["apple  "; "apple juice"; "an apple"])
 ##      @result{} [1; 2]
 ##
 ## strmatch ("apple", ["apple  "; "apple juice"; "an apple"], "exact")
 ##      @result{} [1]
 ## @end group
 ## @end example
 ##
-## @strong{Caution:} @code{strmatch} is scheduled for deprecation.  Use
-## @code{strncmp} (normal case), or @code{strcmp} (@qcode{"exact"} case), or
-## @code{regexp} in all new code.
 ## @seealso{strfind, findstr, strcmp, strncmp, strcmpi, strncmpi, find}
 ## @end deftypefn
 
 ## Author: Paul Kienzle, Alois Schloegl
 ## Adapted-by: jwe
 
+## Deprecated in version 4.2
+
 function idx = strmatch (s, A, exact)
 
+  persistent warned = false;
+  if (! warned)
+    warned = true;
+    warning ("Octave:deprecated-function",
+             "strmatch is obsolete and will be removed from a future version of Octave, please use strncmp, strcmp, or regexp instead");
+  endif
+
   if (nargin < 2 || nargin > 3)
     print_usage ();
   endif
 
   if (! ischar (s) || (! isempty (s) && ! isvector (s)))
     error ("strmatch: S must be a string");
   elseif (! (ischar (A) || iscellstr (A)))
     error ("strmatch: A must be a string or cell array of strings");
diff --git a/scripts/strings/module.mk b/scripts/strings/module.mk
--- a/scripts/strings/module.mk
+++ b/scripts/strings/module.mk
@@ -18,17 +18,16 @@ strings_FCN_FILES = \
   strings/ostrsplit.m \
   strings/regexptranslate.m \
   strings/rindex.m \
   strings/str2num.m \
   strings/strcat.m \
   strings/strchr.m \
   strings/strjoin.m \
   strings/strjust.m \
-  strings/strmatch.m \
   strings/strsplit.m \
   strings/strtok.m \
   strings/strtrim.m \
   strings/strtrunc.m \
   strings/substr.m \
   strings/untabify.m \
   strings/validatestring.m
 
